tests/unit/test_github_enterprise.py

```diff
@@ -0,0 +1,134 @@
+from github3.github import GitHubEnterprise
+import github3
+
+from .helper import UnitHelper
+
+example_data = {
+    "login": "octocat",
+    "id": 1,
+    "avatar_url": "https://github.com/images/error/octocat_happy.gif",
+    "gravatar_id": "somehexcode",
+    "url": "https://api.github.com/users/octocat",
+    "html_url": "https://github.com/octocat",
+    "followers_url": "https://api.github.com/users/octocat/followers",
+    "following_url": ("https://api.github.com/users/octocat/following"
+                      "{/other_user}"),
+    "gists_url": "https://api.github.com/users/octocat/gists{/gist_id}",
+    "starred_url": ("https://api.github.com/users/octocat/starred"
+                    "{/owner}{/repo}"),
+    "subscriptions_url": "https://api.github.com/users/octocat/subscriptions",
+    "organizations_url": "https://api.github.com/users/octocat/orgs",
+    "repos_url": "https://api.github.com/users/octocat/repos",
+    "events_url": "https://api.github.com/users/octocat/events{/privacy}",
+    "received_events_url": ("https://api.github.com/users/octocat/"
+                            "received_events"),
+    "type": "User",
+    "site_admin": False,
+    "name": "monalisa octocat",
+    "company": "GitHub",
+    "blog": "https://github.com/blog",
+    "location": "San Francisco",
+    "email": "octocat@github.com",
+    "hireable": False,
+    "bio": "There once was...",
+    "public_repos": 2,
+    "public_gists": 1,
+    "followers": 20,
+    "following": 0,
+    "created_at": "2008-01-14T04:33:35Z",
+    "updated_at": "2008-01-14T04:33:35Z"
+}
+
+base_url = 'https://ghe.example.com/'
+
+
+def _build_url(_, *args, **kwargs):
+    return github3.session.GitHubSession().build_url(
+        base_url=base_url + 'api/v3', *args, **kwargs)
+
+
+class TestGitHubEnterprise(UnitHelper):
+    described_class = GitHubEnterprise
+    example_data = base_url
+
+    def create_instance_of_described_class(self):
+        instance = self.described_class(self.example_data)
+        instance.session = self.session
+
```

Comment:
not that i disagree with you, but there is no session argument in GitHub class either and since both GitHub and GitHubEnterprise (via GitHub class) inherit from GitHubCore, they both have access to the session object. if it is feasible, can you check if tests pass where you use parent class method create_instance_of_described_class. If you had already done so, then no more exceptions from me with this :)